ext/mysqlnd/mysqlnd_wireprotocol.c

```diff
@@ -292,9 +310,12 @@ mysqlnd_read_packet_header_and_body(MYSQLND_PACKET_HEADER * packet_header,
 {
 	DBG_ENTER("mysqlnd_read_packet_header_and_body");
 	DBG_INF_FMT("buf=%p size=%zu", buf, buf_size);
+	assert(error_info->error_no == 0);
```

Comment:
What is this for? Is this necessary?

Reply:
The following part required `error_info->error_no` are 0 here, is this always true in here?

Reply:
To be honest, I don't know if this is guaranteed, but it seems to me like it should be. Either way, I don't think adding an assert is correct here. Can you remove it and see if it still works?

Reply:
Removing this assert has no effect. I originally considered keeping it in debug mode to catch the impact of other changes in time.

Reply:
I also think this assert is unnecessary.

Reply:
In code like the following, `error_info->error_no` will have a non-zero value:
```
<?php
$conn = mysqli_connect(/**/);
$conn->close();
$conn->execute_query('SELECT 1');
```

Reply:
Got it, I will update this later.

Reply:
Oh, wait a minute. The problem doesn't seem to be here.
It seems like the JIT test is failing, so I'll do some research.

Reply:
`MYSQLND_ERROR_INFO * error_info` can be shared when using persistent connections.
In other words, depending on the processing timing of the processes running in parallel, there is a slight possibility that a value other than `0` may be entered here.

I don't know why such a situation occurred in CI testing.

Reply:
However, it does not reproduce locally even if I run it in parallel. If it's thread-safe, this probably won't happen...

Reply:
I found that there are routes such as `mysqlnd_handle_local_infile` that call `mysqlnd_read_packet_header_and_body` with an error set. Therefore, Assert should not be done.

Reply:
Thanks for the deep analysis.